analyzer/src/main/kotlin/managers/Conan.kt

```diff
@@ -144,8 +147,18 @@ class Conan(
         stashDirectories(directoryToStash).use {
             configureRemoteAuthentication(conanConfig)
 
+            // TODO: Support lockfiles which are located in a different directory than the definition file.
+            val lockfileName = analyzerConfig.packageManagers?.get(managerName)?.options?.get("lockfileName")
+            requireLockfile(workingDir) { lockfileName?.let { hasLockFile(workingDir.resolve(it).path) } ?: false }
+
             val jsonFile = createOrtTempDir().resolve("info.json")
-            run(workingDir, "info", definitionFile.name, "--json", jsonFile.absolutePath, *DUMMY_COMPILER_SETTINGS)
+            if (!analyzerConfig.allowDynamicVersions && lockfileName != null) {
+                verifyLockfileBelongsToProject(workingDir, lockfileName)
+                run(workingDir, "info", definitionFile.name, "-l", lockfileName, "--json", jsonFile.absolutePath)
+            } else {
+                run(workingDir, "info", definitionFile.name, "--json", jsonFile.absolutePath, *DUMMY_COMPILER_SETTINGS
+                )
+            }
```

Comment:
If `allowDynamicVersion == true` and the lockfile is set it should still be used. Best reuse the `requireLockfile` function from `PackageManager`, see e.g. `Pub` for how it is used, and change the logic like that:
```kotlin
requireLockfile(...)

if (lockfileName != null) {
  run(...)
} else {
  run(...)
}
```